src/execution/compiler/operator/sort_translator.cpp

```diff
@@ -209,13 +217,21 @@ ast::FunctionDecl *SortTranslator::GenerateEndTLMergeHookFunction() const {
   auto ret_type = codegen->BuiltinType(ast::BuiltinType::Kind::Nil);
   FunctionBuilder builder(codegen, parallel_endtlmerge_hook_fn_, std::move(params), ret_type);
   {
+    brain::ExecutionOperatingUnitType build_merge_ou_type;
+    ast::Expr *cardinality_val;
+    if (const auto &plan = GetPlanAs<planner::OrderByPlanNode>(); plan.HasLimit()) {
+      build_merge_ou_type = brain::ExecutionOperatingUnitType::PARALLEL_SORT_TOPK_MERGE_STEP;
+      cardinality_val = codegen->Const32(plan.GetOffset() + plan.GetLimit());
+    } else {
+      build_merge_ou_type = brain::ExecutionOperatingUnitType::PARALLEL_SORT_MERGE_STEP;
+      cardinality_val = codegen->MakeExpr(override_value);
+    }
```

Comment:
For parallel merge, it might make sense to just record the `override_value` as the cardinality. Merging parallel sorters does not actually rely on the `LIMIT k`.

Reply:
Changed. But I do have a confusion about the parallel hooks thing as I mentioned in Slack.